nitro-protocol/package.json

```
@@ -1,12 +1,19 @@
 {
   "name": "@statechannels/nitro-protocol",
   "version": "2.0.0-alpha.0",
-  "description": "Smart contracts and typescript libraries for nitro v2",
-  "main": "index.js",
+  "author": "statechannels.org",
+  "bugs": "https://github.com/statechannels/go-nitro/issues",
+  "description": "Smart contracts and typescript libraries for nitro state channel protocol.",
+  "main": "lib/src/index.js",
+  "homepage": "https://github.com/statechannels/go-nitro",
+  "repository": "statechannels/go-nitro.git",
   "files": [
-    "/contracts/**/*.sol"
+    "contracts",
+    "dist"
   ],
+  "types": "dist/src/index.d.ts",
   "scripts": {
+    "build:typescript": "rm -rf dist; npx tsc",
```

Does this work? Pins the tsc version against the semver from devDependencies.

```suggestion
    "build:typescript": "rm -rf dist; tsc",
```

As far as I know the script runners should have access to binaries belonging to to these dependencies.

---

... similar question for much of the `npx` usage.

---

I think you are right. The thing I wanted to avoid is using a globally installed `tsc`, which is what happens if you run it from the command line without `npx`.

---

I'm 100% ready to be wrong about this, but I believe (hope?) that, eg, `npm run someScript` or `yarn someScript` takes from available binaries in `node_modules` before using globally installed binaries.